### PR TITLE
[AVAX] Merge Cortina 8 (v1.10.8)

### PR DESCRIPTION
**Pull request by evlekht**
## Why this should be merged
This PR contains changes from ava-labs/avalanchego between cortina-7 and cortina-8.
https://github.com/ava-labs/avalanchego/releases/tag/v1.10.8
### Conflicts
- .github/workflows/debian/template/control
- RELEASES.md
- genesis/genesis_local.json
- genesis/genesis_test.go
- go.sum
- scripts/build_test.sh
- scripts/mocks.mockgen.txt
- version/compatibility.json
- version/constants.go
- version/string.go
- vms/platformvm/state/mock_state.go
- vms/platformvm/state/state.go
#### Updated genesis IDs
Caminogo doesn't have avax mainnet and fuji genesis configs, so when `GetConfig` is called with one of those ids, it returns local genesis with a modified networkID.

`genesis/genesis_local.json` start time was updated in Cortina 8. 

Because of these two things, all three genesis hashes had changed: avax mainnet, fuji and local.
#### PVM block-builder/network
Starting from cortina 8, `network.blockBuilder` is `*builder`, not a `Builder` interface. Cause of that, we no longer can put caminoBuilder there, so we're directly referencing avax block builder that is wrapped in camino builder.
#### gomock.Controller.Finish() removal
Starting from go 1.14, if controller is created with `*testing.T`, its no longer needed to call `.Finish()` on controller.
See https://github.com/ava-labs/avalanchego/pull/1803
This PR removes similar code for consistency.
## How this was tested
With already existing unit-tests and e2e tests.

### CODE REVIEW

File: vms/platformvm/txs/executor/standard_tx_executor.go [hunk 1]
```diff
@@ -309,6 +320,14 @@ func (e *StandardTxExecutor) AddValidatorTx(tx *txs.AddValidatorTx) error {
...
 
+	if e.Config.PartialSyncPrimaryNetwork && tx.Validator.NodeID == e.Ctx.NodeID {
+		e.Ctx.Log.Warn("verified transaction that would cause this node to become unhealthy",
+			zap.String("reason", "primary network is not being fully synced"),
+			zap.Stringer("txID", txID),
+			zap.String("txType", "addValidator"),
+			zap.Stringer("nodeID", tx.Validator.NodeID),
+		)
+	}
 	return nil
```
Comment (knikos) [inline vms/platformvm/txs/executor/standard_tx_executor.go:331]: with regards to this warning log and taking into consideration https://github.com/ava-labs/avalanchego/pull/1769 and the rationale behind this logs, I believe we should add a similar warning log in our `vms/platformvm/txs/executor/camino_tx_executor.go` and more specifically in the AddValidatorTx implementation.

File: api/admin/service_test.go [hunk 2]
```diff
@@ -100,7 +98,6 @@ func TestLoadVMsGetAliasesFails(t *testing.T) {
 	require := require.New(t)
 
 	resources := initLoadVMsTest(t)
-	defer resources.ctrl.Finish()
```
Comment (knikos) [inline api/admin/service_test.go:103]: we should refactor our test cases to reflect this change according to https://github.com/ava-labs/avalanchego/pull/1803. I guess we should create a different issue for that to keep this PR as minimal as possible

File: .github/workflows/test.e2e.persistent.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Git checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: '1.19'
```
Comment (knikos) [inline .github/workflows/test.e2e.persistent.yml:25]: any reason we ignore this practice from avax to run golang with latest 1.19 patch 
```yaml
          go-version: '~1.19.12'
          check-latest: true
 ```
 That applies also to all other build and test workflows as well as the static-analysis.yml

File: tests/fixture/testnet/config.go [hunk 4]
```diff
@@ -273,7 +273,16 @@ func (nc *NodeConfig) EnsureNodeID() error {
...
+
+	nodeIDBytes, err := secp256k1.RecoverSecp256PublicKey(tlsCert.Leaf)
+	if err != nil {
+		return fmt.Errorf("failed to recover nodeID bytes from tls cert: %w", err)
+	}
+	nodeID, err := ids.ToNodeID(nodeIDBytes)
+	if err != nil {
+		return fmt.Errorf("failed to parse nodeID bytes: %w", err)
+	}
+	nc.NodeID = nodeID
```
Comment (knikos) [inline tests/fixture/testnet/config.go:282]: how come not just call  peer.CertToID(tlsCert.Leaf) directly?

File: api/admin/service_test.go (hunk 2 above)
Comment (evlekht) [reply api/admin/service_test.go:103]: Created issue
https://github.com/chain4travel/camino-private-issues/issues/366

File: .github/workflows/test.e2e.persistent.yml (hunk 3 above)
Comment (evlekht) [reply .github/workflows/test.e2e.persistent.yml:25]: Just consistency - we didn't change form 1.19 to 1.19.6 before, so I'm keeping it this way with assumption that we had a reason for this. I'd prefer to consciously revisit this topic, when we'll do pass on workflows/scripts after cortina merges.

File: vms/platformvm/txs/executor/standard_tx_executor.go (hunk 1 above)
Comment (evlekht) [reply vms/platformvm/txs/executor/standard_tx_executor.go:331]: good point, added

**Review by knikos — APPROVED** [review]

**Pull request merged**